### DISCUSSION TITLE
Allow SocketAddr-syntax in ip_header

### DISCUSSION THREAD

**Discussion by olback** [rwf2/Rocket#2772] [Ideas]
Hi 👋 

I tried deploying a Rocket application behind a IIS Reverse Proxy today but I ran into some trouble.

IIS sets the connecting IP in `x-forwarded-for` header but also includes the client port, like this:
`x-forwarded-for: 10.1.1.6:61234`. This makes the IP "invalid" in the eyes of the `IpAddr` parser.
Would it make sense to allow this syntax as well? Is there a scenario where the port number is useful to someone implementing a Rocket application? Maybe adding a `Request::client_socker_addr()` method?

I did figure out how to make IIS only include the IP in the `x-forwarded-for` header but I think It'd be nice if it worked out of the box.

**Comment by SergioBenitez** [comment]
I'm a bit confused here. As far as I'm aware, Rocket doesn't provide any mechanisms intended for use with `X-Forwarded-For`. It _does_ contain types that use `X-Real-Ip`, but this is a different header with different semantics and security implications - the two should not be conflated. Can you clarify your concern in light of this?
